fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/jaas/JWTLoginModule.java

```diff
@@ -98,4 +100,15 @@ protected JWTCallerPrincipal validate(JWTCredential jwtCredential) throws ParseE
         JWTCallerPrincipal callerPrincipal = factory.parse(jwtCredential.getBearerToken(), jwtCredential.getAuthContextInfo());
         return callerPrincipal;
     }
+
+    @Override
+    protected Group[] getRoleSets() throws LoginException {
+        if (options.containsKey("rolesProperties")) {
+            return super.getRoleSets();
+        } else {
+            Principal group =  subject.getPrincipals().stream()
+                .filter(p -> p instanceof Group && "Roles".equals(p.getName())).findFirst().get();
+            return new Group[] {(Group)group};
```

Comment:
Is this exactly the same as what would happen if I specified an empty file in `rolesProperties`?

Reply:
JWTLoginModule sets the group in commit(). The proposed code will return this group bypassing the super.getRolesSet() only if no properties file (empty or not) is set. If it is an empty file, then RoleMappingLoginModule itself should be OK with that (processRoles()) though I guess the RBAC won't be enforced, it is a diff issue though

Reply:
I want to be very cautious, as this is security-related.

Basically, if we want to make the `rolesProperties` option optional, then when it's missing, it needs to behave exactly like if it was set to an empty file.

Reply:
I run the TCK with empty roles properties file and it worked, so in both cases (with the empty properties and the proposed code) 
https://github.com/MicroProfileJWT/microprofile-jwt-auth/blob/master/tck/src/test/resources/Token1.json#L14
is used as the roles, so I was wrong with my prev comment.
As far as the test is concerned, I can at least duplicate the current TCK and test without the properties, but I wonder if a better option exists :-)

Reply:
Just looked at `RoleMappingLoginModule` and indeed this implementation looks like it will do exactly what would happen if I configured an empty file for `rolesProperties`. With one exception: if `subject.getPrincipals()` doesn't contain a `Group` named `Roles`, the code in `RoleMappingLoginModule` would return `null`, while this code will throw `NoSuchElementException`. It's probably unlikely, but...

Reply:
Hi Ladislav, JWT login module guarantees Group Roles exists since the "groups" is currently a required element so it adds it in its commit

Reply:
Ah, it seems `getRoleSets` is guaranteed to be called only in `super.commit`. OK then, thanks for your patience! :-)

Reply:
I think this is ok.